lib/services/wallet_connect_service.dart

```diff
@@ -0,0 +1,613 @@
+import 'package:flutter/material.dart';
+import 'package:logging/logging.dart';
+import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';
+import 'package:zenon_syrius_wallet_flutter/blocs/blocs.dart';
+import 'package:zenon_syrius_wallet_flutter/blocs/wallet_connect/wallet_connect_pairings_bloc.dart';
+import 'package:zenon_syrius_wallet_flutter/blocs/wallet_connect/wallet_connect_sessions_bloc.dart';
+import 'package:zenon_syrius_wallet_flutter/main.dart';
+import 'package:zenon_syrius_wallet_flutter/model/model.dart';
+import 'package:zenon_syrius_wallet_flutter/utils/functions.dart';
+import 'package:zenon_syrius_wallet_flutter/utils/utils.dart';
+import 'package:zenon_syrius_wallet_flutter/widgets/main_app_container.dart';
+import 'package:zenon_syrius_wallet_flutter/widgets/reusable_widgets/dialogs.dart';
+import 'package:zenon_syrius_wallet_flutter/widgets/reusable_widgets/icons/link_icon.dart';
+import 'package:znn_sdk_dart/znn_sdk_dart.dart';
+
+class WalletConnectService {
+  WalletConnectService._internal();
+
+  factory WalletConnectService() => _instance;
+
+  static final WalletConnectService _instance =
+      WalletConnectService._internal();
+
+  final List<int> _idSessionsApproved = [];
+
+  late Web3Wallet _wcClient;
+  late BuildContext _context;
+
+  final List<SessionData> dAppsActiveSessions = [];
+
+  set context(BuildContext context) => _context = context;
+
+  final _walletLockedError = WalletConnectError(
+    code: 9000,
+    message: 'Wallet is locked',
+  );
+
+  Future<void> initClient() async {
+    _wcClient = await Web3Wallet.createInstance(
```

Comment:
Please wrap this in a conditional statement. The application fails to run if the kWcProjectId is blank.

Example
```
if (kWcProjectId.isEmpty) {
      Logger('WalletConnectService').log(Level.INFO,
          'Failed to initialize WalletConnectService: undefined project id');
    } else {
      _wcClient = await Web3Wallet.createInstance(
      ...
    } 
}
```